dlup/cli/mask.py

```diff
@@ -0,0 +1,197 @@
+            index = int(index)
+            if index == 0:
+                raise argparse.ArgumentTypeError(f"0 is not a proper index. Needs to be at least 1.")
+            index_map[index] = name.strip()
+
+    polygons = dataset_to_polygon(dataset, index_map=index_map, num_workers=args.num_workers, scaling=scaling)
+    wsi_annotations = [
+        WsiSingleLabelAnnotation(
+            label=label,
+            type=AnnotationType.POLYGON,
+            coordinates=[Polygon(polygons[label], label=label)],
+        )
+        for label in polygons
+    ]
+
+    slide_annotations = WsiAnnotations(wsi_annotations)
+
+    if not args.separate:
+        with open(output_filename, "w") as f:
+            json.dump(slide_annotations.as_geojson(split_per_label=False), f, indent=2)
+    else:
+        jsons = slide_annotations.as_geojson(split_per_label=True)
+        for label, json_dict in jsons:
+            suffix = output_filename.suffix
+            name = output_filename.with_suffix("").name
+            new_name = name + "-" + label
+            new_filename = (output_filename.parent / new_name).with_suffix(suffix)
+            with open(new_filename, "w") as f:
+                json.dump(json_dict, f, indent=2)
+
+
+def register_parser(parser: argparse._SubParsersAction):
+    """Register mask commands to a root parser."""
+    wsi_parser = parser.add_parser("mask", help="WSI mask parser")
+    wsi_subparsers = wsi_parser.add_subparsers(help="WSI mask subparser")
+    wsi_subparsers.required = True
+    wsi_subparsers.dest = "subcommand"
+
+    # Tile a slide and save the tiles in an output folder.
+    mask_parser = wsi_subparsers.add_parser(
+        "mask-to-polygon", help="Convert a WSI defining a mask to a GeoJSON polygon."
+    )
+    mask_parser.add_argument(
+        "MASK_FN",
+        type=file_path,
+        help="Filename of the mask. If `--separate` is set, will create a label <MASK_FN>-<label>.json",
+    )
+    mask_parser.add_argument(
+        "OUTPUT_FN",
+        type=lambda x: file_path(x, need_exists=False),
+        help="Output filename. Will create all parent directories if needed.",
+    )
+    mask_parser.add_argument(
+        "--labels",
+        type=str,
+        help="Comma-separated integer value in the mask, name. E.g. specimen=1,tumor=2,...",
+    )
+    mask_parser.add_argument(
+        "--silent",
+        action="store_true",
+        help="If set, will not show progress bar.",
+    )
+    mask_parser.add_argument(
+        "--separate",
+        action="store_true",
+        help="If set, save labels separately.",
+    )
+    mask_parser.add_argument(
+        "--tile-size",
+        type=int,
+        default=1024,
+        help="Size of the generated tiles.",
+    )
+    mask_parser.add_argument(
+        "--tile-overlap",
+        type=int,
+        default=128,
+        help="Number of overlapping pixels between tiles.",
+    )
```

Comment:
when/how do we need tile size and overlap? 

Do we generate tiles?

I thought we took a `mask.tiff` and transformed them to geojson object. I don't understand why we need to generate tiles.